admin/cluster_setup.py

```diff
@@ -79,6 +82,24 @@ def postOptions(self):
                 "Purpose may have only alphanumeric symbols and dash. " +
                 "Found {!r}".format('purpose')
             )
+
+        if self['cert-directory']:
+            # The path should not exist or it should be an empty directory.
+            cert_path = FilePath(self['cert-directory'])
+            if cert_path.exists():
+                if not cert_path.isdir():
+                    raise UsageError(
+                        "{} is not a directory".format(self['cert-directory'])
+                    )
+                if cert_path.listdir():
+                    raise UsageError(
+                        "{} is not empty".format(self['cert-directory'])
+                    )
+            else:
+                cert_path.makedirs()
```

Comment:
If `makedirs()` fails, then we should report this to the user. For example, if they don't have permission to create that directory path.